src/shogun/mathematics/Statistics.cpp

```diff
@@ -814,6 +814,171 @@ SGMatrix<float64_t> CStatistics::sample_from_gaussian(SGVector<float64_t> mean,
 	return S;
 }
 
+CStatistics::SigmoidParamters CStatistics::fit_sigmoid(SGVector<float64_t> scores, SGVector<float64_t> labels)
+{
+	SG_SDEBUG("entering CStatistics::fit_sigmoid()\n")
+
+	REQUIRE(scores.vector, "CStatistics::fit_sigmoid() requires "
+			"scores vector!\n");
+
+	/* count prior0 and prior1 if needed */
+	int32_t prior0=0;
+	int32_t prior1=0;
+	SG_SDEBUG("counting number of positive and negative labels\n")
+	{
+		for (index_t i=0; i<scores.vlen; ++i)
```

Comment:
isn't it enough to count the one of them as the other can be simply deducted?
i would use something like `std::count_if` or `std::transform`

Reply:
Yes, counting one of them would be enough. I using `std::count_if` would indeed be neat.